data-mapper-logstash/swarm.sh

```diff
@@ -1,82 +1,131 @@
 #!/bin/bash
 
-# Arguments
-readonly ACTION=$1
-readonly MODE=$2
+declare ACTION=""
+declare MODE=""
+declare COMPOSE_FILE_PATH=""
+declare UTILS_PATH=""
+declare service_name=""
 
-readonly LOGSTASH_DEV_MOUNT=$LOGSTASH_DEV_MOUNT
+function init_vars() {
+  ACTION=$1
+  MODE=$2
 
-COMPOSE_FILE_PATH=$(
-  cd "$(dirname "${BASH_SOURCE[0]}")" || exit
-  pwd -P
-)
+  COMPOSE_FILE_PATH=$(
+    cd "$(dirname "${BASH_SOURCE[0]}")" || exit
+    pwd -P
+  )
 
-# Import libraries
-ROOT_PATH="${COMPOSE_FILE_PATH}/.."
-. "${ROOT_PATH}/utils/config-utils.sh"
-. "${ROOT_PATH}/utils/docker-utils.sh"
-. "${ROOT_PATH}/utils/log.sh"
+  UTILS_PATH="${COMPOSE_FILE_PATH}/../utils"
 
-inject_pipeline_elastic_hosts() {
+  service_name="data-mapper-logstash"
+
+  if [[ "${NODE_MODE}" == "cluster" ]]; then
+    export LOGSTASH_YML_CONFIG="logstash-logstash.cluster.yml"
+  else
+    export LOGSTASH_YML_CONFIG="logstash-logstash.yml"
+  fi
+
+  readonly ACTION
+  readonly MODE
+  readonly COMPOSE_FILE_PATH
+  readonly UTILS_PATH
+  readonly service_name
+}
+
+# shellcheck disable=SC1091
+function import_sources() {
+  source "${UTILS_PATH}/docker-utils.sh"
+  source "${UTILS_PATH}/config-utils.sh"
+  source "${UTILS_PATH}/log.sh"
+}
+
+function dev_mount_logstash() {
+  if [[ "$LOGSTASH_DEV_MOUNT" == "true" ]]; then
+    if [[ -z $LOGSTASH_PACKAGE_PATH ]]; then
+      log error "LOGSTASH_PACKAGE_PATH environment variable not specified. Please specify LOGSTASH_PACKAGE_PATH as stated in the README."
+      exit 1
+    fi
+
+    log info "Attaching dev mount file"
+    logstash_dev_mount_compose_filename="docker-compose.dev-mnt.yml"
```

Comment:
`logstash_dev_mount_compose_filename` is not passed through as an arg to this function, and it is used in functions outside of dev_mount_logstash() scope. Please declare the variable globally

Reply:
done :)